pandas/core/internals/blocks.py

```diff
@@ -2027,6 +2027,32 @@ def _unstack(self, unstacker, fill_value, new_placement):
         ]
         return blocks, mask
 
+    def replace(
+        self,
+        to_replace,
+        value,
+        inplace: bool = False,
+        regex: bool = False
+    ):
+        """
+        replace the to_replace value with value, regex is not supported by super class
+        when regex is required ObjectBlock replace method is called
+        """
+        inplace = validate_bool_kwarg(inplace, "inplace")
+        regex = validate_bool_kwarg(regex, "regex")
+        if regex:
+            dtype = self.values.dtype
+            block = self.astype(object)
+            if not inplace:
+                return [
+                    b.astype(dtype)
+                    for b in block.replace(to_replace, value, inplace, regex)
+                ]
+            block.replace(to_replace, value, inplace, regex)
+            return block.astype(dtype)
```

Comment:
We recently implement Block._replace_regex.  can you see if we can avoid the object-dtype cast (ideally even this whole method) by using that?

Reply:
replace_regex is using np.vectorize function internally which gives an error when using not object dtype hence using replace_regex after typecasting.
f = np.vectorize(re_replacer, otypes=[values.dtype])